src/services/api/activity-log/activity.log.builder.service.ts

```diff
@@ -0,0 +1,165 @@
+import { IActivity } from '@platform/activity';
+import { ActivityEventType } from '@common/enums/activity.event.type';
+import { IActivityLogEntryMemberJoined } from '@services/api/activity-log/dto/activity.log.dto.entry.member.joined.interface';
+import { IActivityLogEntryCalloutPublished } from '@services/api/activity-log/dto/activity.log.dto.entry.callout.published';
+import { IActivityLogEntryCalloutCardCreated } from '@services/api/activity-log/dto/activity.log.dto.entry.callout.card.created';
+import { IActivityLogEntryCalloutCanvasCreated } from '@services/api/activity-log/dto/activity.log.dto.entry.callout.canvas.created';
+import { IActivityLogEntryChallengeCreated } from '@services/api/activity-log/dto/activity.log.dto.entry.challenge.created';
+import { IActivityLogEntryOpportunityCreated } from '@services/api/activity-log/dto/activity.log.dto.entry.opportunity.created';
+import { IActivityLogEntryCalloutCardComment } from '@services/api/activity-log/dto/activity.log.dto.entry.callout.card.comment';
+import { IActivityLogEntryCalloutDiscussionComment } from '@services/api/activity-log/dto/activity.log.dto.entry.callout.discussion.comment';
+import { IUser } from '@src/domain';
+import { ActivityService } from '@platform/activity/activity.service';
+import { UserService } from '@domain/community/user/user.service';
+import { CalloutService } from '@domain/collaboration/callout/callout.service';
+import { AspectService } from '@domain/collaboration/aspect/aspect.service';
+import { CanvasService } from '@domain/common/canvas/canvas.service';
+import { ChallengeService } from '@domain/challenge/challenge/challenge.service';
+import { OpportunityService } from '@domain/collaboration/opportunity/opportunity.service';
+import { CommunityService } from '@domain/community/community/community.service';
+
+interface ActivityLogBuilderFunction<TypedActivityLogEntry> {
+  (rawActivity: IActivity): Promise<TypedActivityLogEntry>;
+}
+
+interface IActivityLogBuilder {
```

Comment:
I would prefer to enforce exhausting all members of `ActivityEventType` to reduce potential mistakes.

Reply:
One type per file. That is what is done everywhere else on the server and lets keep to that pattern unless we explicitly decide to move away from this. So please create an activity.log.builder.interface file with this type in it. Also the type name should start with "I" to designate an interface.